libraries/omnitruck_handler.rb

```diff
@@ -57,12 +41,27 @@ def handle_uninstall
     end
 
     def configure_version(installer)
-      installer_script_path = File.join(Chef::Config[:file_cache_path], 'installer.sh')
       install_command_resource = "install-#{new_resource.product_name}-#{new_resource.version}"
 
       file installer_script_path do
         content installer.install_command
-        notifies :run, "execute[#{install_command_resource}]", :immediately
+        if windows?
+          notifies :run, "powershell_script[#{install_command_resource}]", :immediately
+        else
+          notifies :run, "execute[#{install_command_resource}]", :immediately
+        end
+      end
+
+      powershell_script install_command_resource do
+        # We pass the install code directly, but still depend upon the file to
+        # change before executing the install
+        code installer.install_command
```

Comment:
Note here I'm passing the code string directly.  We could pass the file path as well.  It makes the initial file creation seem a bit pointless if we are converging the file to notify powershell_script.  Is there another way that the resource could check for a "code change" before running it?  Otherwise it may be more clear to pass the file path instead.